lambdas/data-migration2/src/index.ts

```diff
@@ -5,43 +5,71 @@ import { migrateExecutions } from './executions';
 import { migrateGranulesAndFiles } from './granulesAndFiles';
 import { migratePdrs } from './pdrs';
 
+import { GranuleDynamoSearchParams } from './types';
+
 const logger = new Logger({ sender: '@cumulus/data-migration2' });
+
+type allowedMigrations = 'granules' | 'executions' | 'pdrs';
 export interface HandlerEvent {
   env?: NodeJS.ProcessEnv
+  granuleSearchParams?: GranuleDynamoSearchParams
+  migrationsList?: allowedMigrations[]
 }
 
-export const handler = async (event: HandlerEvent): Promise<string> => {
+export const handler = async (
+  event: HandlerEvent
+): Promise<string> => {
   const env = event.env ?? process.env;
+  const migrationsToRun = event.migrationsList ?? ['executions', 'granules', 'pdrs'];
 
   const knex = await getKnexClient({ env });
 
   try {
-    const executionsMigrationSummary = await migrateExecutions(env, knex);
-    const granulesAndFilesMigrationSummary = await migrateGranulesAndFiles(env, knex);
-    const pdrsMigrationSummary = await migratePdrs(env, knex);
-
-    const summary = `
+    let summary = `
       Migration summary:
+    `;
+
+    if (migrationsToRun.includes('executions')) {
```

Comment:
Adding the ability to run only certain parts of data-migration2 seemed like a necessity for re-run scenarios where users only want to retry migrating certain failed executions or granules. 

Adding this configurability is a bit dangerous, as if users were to configure data-migration2 to only do `granules` on the **first run**, then they would all fail as they could not refer to the executions that weren't migrated. But overall I think the configurability is desirble and we just need to document appropriately